### PR TITLE
Weather

### LINKED ISSUE #39: Implement Weather Gathering from Internet

### LINKED ISSUE #44: Create Temperature Sensor Basic Hardware Test
Simple test to run check the temperature from the sensor every 5-10 seconds and print to console. Use for calibration and temperature sensor hardware test.

### PR DESCRIPTION
**Pull request by samjwu**
Closes #39 
Closes #44 

Make location/weather API calls (gets sun angle, temp, cloud cover)
Add temp sensor code

### CODE REVIEW

File: controlalgorithm/persistent_data.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            persistent_data_dict = json.load(fp)
+        lat = persistent_data_dict["lat"]
+        lon = persistent_data_dict["lon"]
+        timezone_adjustment = persistent_data_dict["timezone_adjustment"]
+
+    # if not, get the data and create the json file
+    else:
+        persistent_data_dict = dict()
+
+        place_name = input("Enter location or place name: ")
```
Comment (ishchow) [inline controlalgorithm/persistent_data.py]: We can't wait for user input on the RPI code. The RPI code runs entirely headlessly. We should find another way to get the user's location. Or we can just hardcode it for now.

Comment (ishchow) [reply controlalgorithm/persistent_data.py]: Ideally in our system, the only thing that really interacts with the user is the web app. So we either use an API end point to send configuration (ex. timezone) over the very first time the blind is paired or we find some other method that doesn't involve direct user input in the python console.

File: controlalgorithm/persistent_data.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
[the remark this answers is not in the document]
Comment (samjwu) [reply controlalgorithm/persistent_data.json]: added to .gitignore

File: controlalgorithm/persistent_data.py (hunk 1 above)
Comment (samjwu) [reply controlalgorithm/persistent_data.py]: In a production setting, the external application should send this data. @kdehaan 
For the demo, we can hard code this for convenience and time constraints.

**Pull request merged**